assets/js/components/ClusterSettingsPage/ClusterSettingsPage.jsx

```diff
@@ -49,23 +76,41 @@ function ClusterSettingsPage() {
       })
     );
 
-  const saveSelection = (checks) =>
-    dispatch(
-      clusterChecksSelected({
-        clusterID,
-        clusterName,
-        checks,
-      })
-    );
+  const requestExecution = () => {
+    dispatch(executionRequested(clusterID, clusterHosts, selection, navigate));
+  };
 
   return (
     <div className="w-full px-2 sm:px-0">
       <BackButton url={`/clusters/${clusterID}`}>
         Back to Cluster Details
       </BackButton>
-      <PageHeader>
-        Cluster Settings for <span className="font-bold">{clusterName}</span>
-      </PageHeader>
+
+      <div className="flex flex-wrap">
+        <div className="flex w-1/2 h-auto overflow-hidden overflow-ellipsis break-words">
+          <PageHeader>
+            Cluster Settings for{' '}
+            <span className="font-bold">{clusterName}</span>
+          </PageHeader>
+        </div>
+        <div className="flex w-1/2 justify-end">
+          <div className="flex w-fit whitespace-nowrap">
+            <Button type="primary" className="mx-1" onClick={saveSelection}>
```

Comment:
For a consistent ux we could disable also this button while saving the selection.